mod-source-record-manager-server/src/main/java/org/folio/services/MappingRuleServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+@Service
+public class MappingRuleServiceImpl implements MappingRuleService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MappingRuleServiceImpl.class);
+  private static final Charset DEFAULT_RULES_ENCODING = StandardCharsets.UTF_8;
+  private static final String DEFAULT_RULES_PATH = "rules/rules.json";
+  private MappingRuleDao mappingRuleDao;
+
+  public MappingRuleServiceImpl(@Autowired MappingRuleDao mappingRuleDao) {
+    this.mappingRuleDao = mappingRuleDao;
+  }
+
+  @Override
+  public Future<Optional<JsonObject>> get(String tenantId) {
+    return mappingRuleDao.get(tenantId);
+  }
+
+  @Override
+  public Future<Void> saveDefaultRules(String tenantId) {
+    Future<Void> future = Future.future();
+    Optional<String> optionalRules = readResourceFromPath(DEFAULT_RULES_PATH);
+    if (optionalRules.isPresent()) {
+      String rules = optionalRules.get();
+      if (isValidJson(rules)) {
+        mappingRuleDao.save(new JsonObject(rules), tenantId).setHandler(ar -> {
+          if (ar.failed()) {
+            LOGGER.error("Can not save rules for tenant {}", tenantId, ar.cause());
+            future.fail(ar.cause());
+          } else {
+            future.complete();
+          }
+        });
+      } else {
+        String errorMessage = "Can not work with rules in non-JSON format";
+        LOGGER.error(errorMessage);
+        future.fail(new BadRequestException(errorMessage));
+      }
+    } else {
+      String errorMessage = "No default rules found";
+      LOGGER.error(errorMessage);
+      future.fail(errorMessage);
+    }
+    return future;
+  }
+
+  @Override
+  public Future<JsonObject> update(String rules, String tenantId) {
+    Future<JsonObject> future = Future.future();
+    if (isValidJson(rules)) {
+      mappingRuleDao.update(new JsonObject(rules), tenantId).setHandler(future);
+    } else {
+      String errorMessage = "Can not update rules in non-JSON format";
+      LOGGER.error(errorMessage);
+      future.fail(new BadRequestException(errorMessage));
+    }
+    return future;
+  }
+
+  @Override
+  public Future<JsonObject> restore(String tenantId) {
+    Future<JsonObject> future = Future.future();
+    Optional<String> optionalRules = readResourceFromPath(DEFAULT_RULES_PATH);
+    if (optionalRules.isPresent()) {
+      String rules = optionalRules.get();
+      update(rules, tenantId).setHandler(future);
+    } else {
+      String errorMessage = "No rules found in resources";
+      LOGGER.error(errorMessage);
+      future.fail(new InternalServerErrorException(errorMessage));
+    }
+    return future;
+  }
+
+  /**
+   * Returns true if given String is valid JSON
+   *
+   * @param json given string
+   * @return true if given String is valid JSON, false if non-valid
+   */
+  private boolean isValidJson(String json) {
```

Comment:
@RuslanLavrov , @KaterynaSenchenko , @OleksiiKuzminov     please help me to find more useful way to check String if it's in Json format or not